### DISCUSSION TITLE
dev-tools with dynamically added scenes

### DISCUSSION THREAD

**Discussion by chrisk-7777** [excaliburjs/Excalibur#2399] [Q&A] [answered]
Hey hey Excalibur team, 

Before raising an issue or PR I would like to double check if I'm doing something wrong (which is highly probable)

If I add a scene "dynamically", _after_ the game has started, it is not reflected in the scene dropdown in the dev-tools. 
I can see the engine scenes are collated once in dev tools, during the main build here: https://github.com/excaliburjs/dev-tools/blob/main/dev-tools.ts#L194

Is this a bug or am I misusing scenes by adding them post-start?

## Example:

```typescript
export class Game extends Engine {
  private area: Area | null = null;

  public teleport(area: string, tileX: number, tileY: number) {
    // Clean up existing
    if (this.area) {
      this.remove(this.area);
    }

    // Update store
    changeArea(area, tileX, tileY);

    // Update scene
    this.area = new Area();
    this.add('area', this.area);
    this.goToScene('area');
  }
}
```

<img width="415" alt="image" src="https://user-images.githubusercontent.com/55621012/177325100-638498a2-9196-4fce-81d8-5d49cda755dd.png">

## Next

If scenes should only be added once at the start, I might open a PR to the docs under https://excaliburjs.com/docs/scenes#adding-a-scene

If this is an issue with dev-tools, I can look at a PR there instead. Possibly extract the scene pane into its own handler that can be called on update? Or would it be better to listen to some engine event, then update the pane?

**Comment by eonarheim** [comment, marked as answer]
@chrisk-7777 I'd call this a bug in the dev-tools, it's completely reasonable to add a scene dynamically during the course of a game 👍 The dev-tools should handle new scenes being added. 

I think both suggestions are good ways to tackle this, and would support either approaches
1. Check in the dev-tools update for new scenes periodically, this would be the quickest fix most likely.
2. Add a Scene `sceneadded`/`sceneremoved` event to excalibur that can be listened to in the dev-tools (this event might be a useful feature for excalibur in general!)

Apologies, the dev-tools are definitely a minimum viable product. I'm sure there are more than a few quirks in there.

**Reply by chrisk-7777** [reply]
Awesome, thanks Erik. I have a bit of solution going based on (1), its causing a few issues with the picker system but I'll aim to get it fixed up and raised over the weekend. 

Option (2) does sound ideal, but (1) might be a suitable stopgap until I jump into the Excalibur internals

**Reply by chrisk-7777** [reply]
And addressed here: https://github.com/excaliburjs/dev-tools/pull/5

**Reply by eonarheim** [reply]
@chrisk-7777 I've published v0.27.1 of the dev-tools 🎉

**Reply by chrisk-7777** [reply]
Whoo! Thanks Erik
